migrations/versions/fe3b1041ed15_add_bmres_timestamp_index.py

```diff
@@ -0,0 +1,28 @@
+"""add_bmres_timestamp_index
+
+Revision ID: fe3b1041ed15
+Revises: 21519d1e3ddb
+Create Date: 2023-03-01 15:41:25.181769
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'fe3b1041ed15'
+down_revision = '21519d1e3ddb'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_index('benchmark_result_timestamp_index', 'benchmark_result', ['timestamp'], unique=False)
```

Comment:
So the main problem I foresee is that the Arrow `benchmark_result` table has 30 million rows. Creating the index will probably take a while, and will lock out writes in the meantime. This will probably cause a few benchmark builds to fail with 500s from Conbench when trying to POST.

There is the option to use `postgresql_concurrently=True` here, which will create the index concurrently. That will allow for no locks but "takes significantly longer to complete"[[docs](https://www.postgresql.org/docs/current/sql-createindex.html)]. I have no idea what this order of magnitude of time will be. [This SO answer](https://dba.stackexchange.com/a/249784) says that we'll be able to monitor the progress at least.

Reply:
Thanks for the thought. I have pushed a commit to instruct postgres to create the index concurrently. In our case here I don't think it will take much longer than the non-concurrent build because I think in our case the mutation transactions are small / short. The concurrent build can theoretically take foreverish because

>  must wait for all existing transactions that could potentially modify or use the index to terminate

This becomes a challenge in DBs with continuous write pressure which I assume/suspect we do not have (the DB I hope spends the majority of its lifetime _not_ writing).

-----

The rest of my comment is not too relevant anymore for this PR because we now try the concurrent index build. But because timescale considerations are always interesting to discuss and a powerful tool to make decisions, I'd love to continue the thought experiment:

> will probably take a while, and will lock out writes in the meantime.

It would be cool if we had an idea about the ballpark. One minute? 10 minutes? 1 hour?

> with 500s from Conbench when trying to POST.

We should have a pretty clear idea about how long recommended client tooling (tooling that _we_ build) attempts to retry, before it drops data and moves on. In similar systems I have always advocated for and also built an explicit, tested design choice: "we want to survive a micro outage / maintenance window of N minutes length; that is our promise". Such a deliberate goal makes it easy to develop/test, and it also makes it easy to communicate guarantees to users.

Given knowledge about those timescales we could make a rather credible estimation: would the non-concurrent index build lead to 'data loss' or not?

For example, if the index build is expected to take ~10 minutes, and client tooling is known to retry for up to one hour, we would have a pretty clear perspective.